qtconsole/tests/test_aaconsole_widget.py

```diff
@@ -10,12 +10,139 @@
 from qtconsole.qt import QtCore, QtGui
 from qtconsole.qt_loaders import load_qtest
 from qtconsole.console_widget import ConsoleWidget
+from qtconsole.qtconsoleapp import JupyterQtConsoleApp
 from . import no_display
 
 
+SHELL_TIMEOUT = 20000
 QTest = load_qtest()
 
 
+@pytest.fixture
+def qtconsole(qtbot):
+    """Qtconsole fixture."""
+    # Create a console
+    console = JupyterQtConsoleApp()
+    console.initialize(argv=[])
+
+    qtbot.addWidget(console.window)
+    console.window.confirm_exit = False
+    console.window.show()
+    return console
+
+
+def test_scroll(qtconsole, qtbot):
+    """
+    Make sure the scrolling works.
+    """
+    window = qtconsole.window
+    shell = window.active_frontend
+    control = shell._control
+    scroll_bar = control.verticalScrollBar()
+
+    # Wait until the console is fully up
+    qtbot.waitUntil(lambda: shell._prompt_html is not None,
+                    timeout=SHELL_TIMEOUT)
+
+    assert scroll_bar.value() == 0
+    # Create a bunch of inputs
+    for i in range(20):
+        with qtbot.waitSignal(shell.executed):
+            QTest.keyClicks(control, 'a = 1')
+            QTest.keyClick(control, QtCore.Qt.Key_Enter,
+                           modifier=QtCore.Qt.ShiftModifier)
```

Comment:
You're still using `QTest.keyClicks` instead of `qtbot.keyClicks`. Please use the later.

Reply:
But only in your new tests, I mean.